### PR TITLE
Alignment of axes & bar chart

### PR DESCRIPTION
**Pull request by MattyK14**
Thank you for contributing a pull request. 

Please ensure that you have signed the [CLA](https://docs.google.com/forms/d/19LpBBjykHPox18vrZvBbZUcK6gQTj7qv1O5hCduAZFU/viewform
).

- [x] I have signed the CLA

### CODE REVIEW

**Comment by marzolfb** [comment]
A lot of this unwinds several axis/offset changes in the previous PR which admittedly might only be addressing behavior using a specific version of dependencies. Out of curiosity what versions of react-native and react-native-svg are you working with?

**Comment by MattyK14** [comment]
I was having the issue on RN 0.38 and 0.39. SVG is 4.4.0

**Comment by marzolfb** [comment]
I'm trying to understand what change you are proposing here. If I take just the bar chart from the example app and compare it before and after this proposed change, I see the following visual changes (running on an Android emulator using RN 0.38 and SVG 4.4.1):

BEFORE
image

AFTER
image

Do you have a different set of data/options other than the example app that illustrates the changes differently?

Do you see roughly the same before and after pictures I do?

Even so, there are certainly differences in the before and after pics above:

- The x-axis labels on the AFTER pic do seem to be more centered with the bars than the labels on the BEFORE pic (a change for the better)
- The y-axis line is quite a bit longer on the AFTER pic vs the BEFORE pic extending well below the origin (and quite different than the behavior of the x-axis line) and extending further above as well (a change in need of improvement)

**Comment by MattyK14** [comment]
In my circumstance the bars of the graph were also shifted with the labels in my before

**Comment by marzolfb** [comment]
I've tried a couple of different times to reproduce what you describe (getting bars and labels shifted in the current source before applying this PR) and I haven't been able to do it. To reproduce your setup, I should be able to:

- Modify package.json in the example dir to upgrade react and react-native (since you indicated above you are using :
```
-    "react": "~15.3.0",
-    "react-native": "^0.35.0",
+    "react": "~15.4.0",
+    "react-native": "^0.38.0",
```
- Clear out all packages for a fresh starting point
```
rm -rf node_modules
```
- Re-install all dependencies
```
npm install
```

- Verify we have the versions we want
```
> npm list react react-native react-native-svg

example@0.0.1 .../example
├── react@15.4.1 
├── react-native@0.38.1 
└─┬ react-native-pathjs-charts@0.0.21
  └── react-native-svg@4.4.1 
```
- Startup the example app
```
react-native run-android
```

When I do all that I don't see what you indicated above. Are you doing something different that I've missed in the steps I listed? 

Also, given that this PR modifies margin/offset values in Axis.js I decided to take a look at how this PR handles variations in the top & left margin values. Just concentrating on the bar chart, I changed the top & left margin values for the bar chart in the example to 100 and 50
```
        options: {
            width: 300,
            height: 300,
            margin: {
-                top: 20,
-                left: 25,
+               top: 100,
+               left: 50,
```

and then I decided to see what this looked like with various versions of Axis.js and Bar.js.

First, I wanted to see what it looked like before the most recent commit (aa934d235b8becdcac4d1692351f6bf2bf713d9a):

image

That's not what we want - that seems like the axis and labels respect the updated margins but the bars and y-axis gridlines do not

Next I tried the current versions of Axis.js and Bar.js:

image

Well, thats better but we are missing the labels. This one slipped through the cracks and there is certainly a need for a PR to clean this up.

Finally, I tried the versions you supplied in this PR and this is what I see:

image

Well, we see the labels now but now the y-axis line is extending way above and way below where we want it to be.

Just for grins, I decided to pair the current version of Axis.js with the version of Bar.js that you have in this PR and I see this:

image

That is what we are looking for! 

In short, before accepting this PR:
- We need to get to the same starting point - either provide me info on how to get to the same starting state you are at where both bars and labels are shifted or you confirm that you are able to get to the same starting point with the instructions I have above
- Assuming you are at the same starting point I am, modify the PR to keep changes to Bar.js but remove changes to Axis.js. (If we get to your starting point, we can work some more on this from there...)

**Comment by MattyK14** [comment]
I'm kind of short on time right now, but reverting back to the original Axis.js and my new version of Bar.js seems to have no visual effect for me (The Y-axis lines extend further than we'd like).

**Comment by marzolfb** [comment]
@MattyK14 What do you show when you do this from the directory you're running your example from:
```
npm list react react-native react-native-pathjs-charts
```

**Comment by marzolfb** [comment]
Closing this out since I haven't had a response to troubleshooting questions for a while. We can reopen if needed later.
